Code.gs

```diff
@@ -99,196 +98,192 @@ function DeleteAllTriggers(){
-
 }
 
 function ConvertToCustomEvent(vevent){
   var event = new Event();
-  event.id = vevent.getFirstPropertyValue('uid');
+  event.id = vevent.getFirstPropertyValue('uid').toLowerCase().replace(/[-w_x@y.z,]+/g, '');
+  event.created = vevent.getFirstPropertyValue('last-modified') || vevent.getFirstPropertyValue('created');
   
   if (descriptionAsTitles)
-    event.title = vevent.getFirstPropertyValue('description') || '';
+    event.summary = vevent.getFirstPropertyValue('description') || '';
   else{
-    event.title = vevent.getFirstPropertyValue('summary') || '';
+    event.summary = vevent.getFirstPropertyValue('summary') || '';
     event.description = vevent.getFirstPropertyValue('description') || '';
   }
   
   if (addOrganizerToTitle){
     var organizer = ParseOrganizerName(vevent.toString());
-
+    
     if (organizer != null)
-      event.title = organizer + ": " + event.title;
+      event.summary = organizer + ": " + event.summary;
   }
   
   event.location = vevent.getFirstPropertyValue('location') || '';
   
   var dtstart = vevent.getFirstPropertyValue('dtstart');
   var dtend = vevent.getFirstPropertyValue('dtend');
+  var duration = vevent.getFirstPropertyValue('duration') || defaultDuration;
   
-  if (dtstart.isDate && dtend.isDate)
-    event.isAllDay = true;
-  
-  if (vtimezone != null)
-    dtstart.zone = new ICAL.Timezone(vtimezone);
+  if (dtstart.isDate && dtend.isDate){
+    event.start = {date: dtstart.toString(),
+                   timeZone: tzid};
+    if (dtend == null){
+      event.end = event.start;
+    }else{
+      event.end = {date: dtend.toString(),
+                   timeZone: tzid};
+    }
+  }else{
+    event.start = {dateTime: dtstart.toString(),
+                   timeZone: tzid};
     
-  event.startTime = dtstart.toJSDate();
-  
-  if (dtend == null)
-    event.endTime = new Date(event.startTime.getTime() + defaultDuration * 60 * 1000);
-  else{
-    if (vtimezone != null)
-      dtend.zone = new ICAL.Timezone(vtimezone);
-      
-    event.endTime = dtend.toJSDate();
+    if (dtend == null){
+      event.end = {dateTime: Utilities.formatDate(new Date(event.start.dateTime + duration * 60 * 1000), tzid, "yyyy-MM-dd\'T\'HH:mm:ss"),
+                   timeZone: tzid};
+    }else{
+      event.end = {dateTime: dtend.toString(),
+                   timeZone: tzid};
+    }
   }
-  
   if (addAlerts){
     var valarms = vevent.getAllSubcomponents('valarm');
     for each (var valarm in valarms){
       var trigger = valarm.getFirstPropertyValue('trigger').toString();
-      event.reminderTimes[event.reminderTimes.length++] = ParseNotificationTime(trigger);
+      event.reminders.overrides[event.reminders.overrides.length++] = ParseNotificationTime(trigger);
     }
   }
-  
+  //RFC allows multiple Rules, Google supports that as well even though it's not displayed in the event details page
+  var recurrenceRules = vevent.getAllProperties('rrule');
+  event.recurrence = [];
```

Comment:
`recurrence` is never used elsewhere in the code. How does this add recurrence support?

Reply:
Rules are added on line 285

Reply:
Right, the `recurrence` array is populated but nothing ever *uses* the recurrence array in order to create a recurring event

Reply:
Recurring events are created automatically when inserting an event with rrules (at least that's the case when using Calendar API). No need to create follow on events manually.

Reply:
I'll have to check that. Note that GAS-ICS-Sync does *not* use the full Google Calendar API, but rather uses the [Google Apps Script](https://developers.google.com/apps-script/reference/calendar/) version

Reply:
I'm aware of that, if changing to the API is not an option for the project my changes most likely are invalid.